File: code/2025/01/28/119. 杨辉三角 II.cpp
class Solution {
  public:
	vector<int> getRow(int n) {
		std::vector<std::vector<int> > ans {{1}};
		for (int i = 1; i <= n; i++) {
			ans.push_back({});
			for (int j = 0; j <= i; j++) {
				if (j == i or j == 0) {
					ans[i].push_back(1);
				} else {
					ans[i].push_back(ans[i - 1][j] + ans[i - 1][j - 1]);
				}
			}
		}
		return ans[n];
	}
};

File: code/2025/03/28/119. 杨辉三角 II.py
class Solution:
	def getRow(self, rowIndex: int) -> List[int]:
		def f(n):
			ans = 1
			for i in range(1, n + 1):
				ans *= i
			return ans
		def C(n, m):
			return f(n) / (f(m) *  f(n - m))
		ans = []
		for i in range(0, rowIndex + 1):
			ans.append(int(C(rowIndex, i)))
		return ans